compiler/main/driver.cpp

```diff
@@ -1338,11 +1338,20 @@ static void postTaskTracking() {
 }
 
 static void postStaticLink() {
-  if (fLinkStyle == LS_STATIC) {
-    if (strcmp(CHPL_TARGET_PLATFORM, "darwin") == 0) {
+  if (!strcmp(CHPL_TARGET_PLATFORM, "darwin")) {
+    if (fLinkStyle == LS_STATIC) {
       USR_WARN("Static compilation is not supported on OS X, ignoring flag.");
       fLinkStyle = fMultiLocaleInterop ? LS_DYNAMIC : LS_DEFAULT;
     }
+
+    //
+    // The default link style translates to static (at least for now), so we
+    // need to account for that when building a multi-locale library or else
+    // we'll get the same errors we do for static linking on `darwin`.
+    //
+    if (fMultiLocaleInterop && fLinkStyle == LS_DEFAULT) {
+      fLinkStyle = LS_DYNAMIC;
+    }
```

Comment:
Can you explain to me why the original strategy for multilocale executables didn't apply to multilocale libraries?  Nothing that was here before seemed like it would have caused the strategy to apply to one and not the other, which makes me wonder if this is the right spot to be modifying.

Reply:
Executables have the following sort of logic when choosing what flags to append in `codegen_makefile`:

```
if (dynamic) then "$(DYNAMIC_FLAG)" else if (static) then "$(STATIC_FLAG)" else "";
```

Whereas libraries in general have the logic:

```
if (dynamic) then "$(DYNAMIC_FLAG)"  else "$(STATIC_FLAG)";
```

So for libraries, `LS_DEFAULT` is effectively equivalent to `LS_STATIC`.

---

If you'd like, I could move this conditional logic further down to `codegen_makefile`, or I could add another pass to `post` in `compiler/main/driver.cpp` which selectively changes `LS_DEFAULT` to `LS_DYNAMIC`/`LS_STATIC` depending on what operating system you are on.

Both of these are a bit more invasive, which is why I opted to go with the above code, though I agree with you it does feel tacked on.

Reply:
Okay, that seems fine